### PR TITLE
Introduce @ff.reintroduce and @Reintroduce

### PR DESCRIPTION
**Pull request by mhdirkse**

### CODE REVIEW

**Comment by mhdirkse** [comment]
TODO:
- [ ] Apply @ff.reintroduce as intended by the issue. Check the XSD and the JSON.
- [ ] Update FRANKDOC.md in the F!F and README.md in the Frank!Doc.

**Comment by mhdirkse** [comment]
This is how the Frank!Doc webapp becomes for page SenderPipe. Inheritance is shown:
image

**Comment by mhdirkse** [comment]
And this is how the XSD will be become for SenderPipe:

```
  <xs:group name="SenderPipeDeclaredChildGroup">
    <xs:sequence>
      <xs:group ref="MessageLogElementGroup" minOccurs="0" maxOccurs="1" />
      <xs:group ref="InputWrapperElementGroup" minOccurs="0" maxOccurs="1" />
      <xs:group ref="InputValidatorElementGroup" minOccurs="0" maxOccurs="1" />
      <xs:group ref="SenderElementGroup" minOccurs="1" maxOccurs="1" />
      <xs:group ref="ListenerElementGroup_2" minOccurs="0" maxOccurs="1" />
    </xs:sequence>
  </xs:group>
  <xs:group name="SenderPipeCumulativeChildGroup">
    <xs:sequence>
      <xs:group ref="SenderPipeDeclaredChildGroup" />
      <xs:group ref="OutputValidatorElementGroup" minOccurs="0" maxOccurs="1" />
      <xs:group ref="OutputWrapperElementGroup" minOccurs="0" maxOccurs="1" />
      <xs:group ref="AbstractPipeDeclaredChildGroup" />
    </xs:sequence>
  </xs:group>
```

**Comment by mhdirkse** [comment]
To use the @Reintroduce annotation, please merge the following F!F PR: https://github.com/ibissource/iaf/pull/3529

**Comment by ricardovh** [comment]
The webapp page does not look correct, the SenderPipe has messageLog, inputWrapper, inputValidator, sender, listener but no outputValidator and outputWrapper. 

I also think that it will confuse people if nested elements are repeated on the page, now the SenderPipe has a list of nested elements, and the MessageSendingPipe from which it inherits has a different list of nested elements. If I look at it as a Frank!Developer, I should not have to care about inheritance, I just want to know for an element which nested elements, attributes and parameters I can set, and ideally each list should be together and only once on the page, but your current print screen makes it look like it will be all over the page with this PR.

**Comment by mhdirkse** [comment]
I had a phone call with @nielsm5. We decided that we will ignore @Reintroduce for the JSON. This will address Ricardo's comment. The @Reintroduce annotation will still have its effect on the XSDs.

**Comment by gvanbrakel** [comment]
I agree with @ricardovh that the screen shot above is confusing for developers, BUT this is in the 'show inheritance' view. The default 'hide inheritance' view is what the developer normally wants to see. There the child elements should be displayed in the right order. 
To enable this, a reintroduced-attribute must be added to the json for the attributes and config-childs that have the annotation.

**Comment by mhdirkse** [comment]
Ik heb Gerrit's opmerking verwerkt.

**Comment by mhdirkse** [comment]
ElementChild may be improved by merging isReintroduced tests with technical override. I will try it out.

File: src/test/java/nl/nn/adapterframework/doc/Reintroduce.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import java.lang.annotation.Documented;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * If set on attribute setter or config child setter, forces attribute or config child to
+ * be declared explicitly with the child. This annotation was introduced to modify the sort
+ * order. Setting this application is only needed if the
```
Comment (gvanbrakel) [inline src/test/java/nl/nn/adapterframework/doc/Reintroduce.java:29]:
```suggestion
 * be declared explicitly with the child. This annotation was introduced to modify the order
 * of appearance. Setting this application is only needed if the
```

File: README.md [hunk 2]
```diff
@@ -272,3 +272,7 @@ The default value is `nl.nn.adapterframework.pipes.SenderPipe` in this example.
 **@Optional:** Java annotation that does the same as JavaDoc tag `@ff.optional`.
 
 **@ff.tag:** Add tag to Java class that is shown in the Frank!Doc webapplication. Add the `@ff.tag` JavaDoc tag in the JavaDoc comment above a class. There are two arguments. The first argument is the tag name and the second argument is the tag value.
+
+**@Reintroduce:** Java annotation meant to change the order of config children or attributes in the XSD. The default order is that inherited element children go after declared element children. To have a different order, override an inherited config child setter or attribute setter and add this annotation. The inherited element child is then sorted with the declared element children. Please note that an inherited element child is also sorted with the declared element children if the setter is overridden with a new JavaDoc comment. This annotation is needed if and only if the only modification of the element child is the sort order.
```
Comment (gvanbrakel) [inline README.md:276]: het sorteren van attributen en config-children is iets interns van de FrankDoc code. Voor de gebruiker is het de volgorde waar in de elementen getoond worden.

**Review by gvanbrakel — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (mhdirkse) [reply README.md:276]: Nu snap ik het punt. Ik heb je voorgestelde verandering gecommit. De tekst die hier gewijzigd wordt is test code; een kopie van de @Reintroduce annotatie die nodig is voor de unit tests. Ik zal kijken of deze verandering ook toegepast kan worden op het origineel in het iaf project.

**Review by gvanbrakel — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (mhdirkse) [reply README.md:276]: Ik heb dit naar het iaf project gekopiëerd maar daarna zag ik geen wijziging in de @Reintroduce Java class. Blijkbaar stond de voorgestelde tekst daar al.

**Pull request merged**